src/pages/api/profile/addresses/index.ts

```diff
@@ -21,6 +22,71 @@ import { authMiddleware } from "@src/middlewares/auth-middleware";
 import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
 import { getProfileAndAddressAccessTokens } from "@src/utils/get-profile-and-address-access-tokens";
 
+const getHandler: Handler = async (request, response) => {
+  const webErrors = {
+    invalidProfileToken: ERRORS_DATA.INVALID_PROFILE_TOKEN,
+    invalidScopes: ERRORS_DATA.INVALID_SCOPES,
+    profileDataNotFound: ERRORS_DATA.PROFILE_DATA_NOT_FOUND,
+    unreachable: ERRORS_DATA.UNREACHABLE,
+  };
+
+  return getTracer().span(
+    "GET /api/profile/addresses handler",
+    async (span) => {
+      const userCookie = await getServerSideCookies<UserCookie>(request, {
+        cookieName: "user-cookie",
+        isCookieSealed: true,
+        cookieSalt: configVariables.cookieSalt,
+      });
+
+      if (!userCookie) {
+        response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+        response.setHeader("location", "/");
+        response.end();
+        return;
+      }
+
+      const { addressAccessToken } = await getProfileAndAddressAccessTokens(
+        userCookie.access_token,
+      );
+
+      span.setDisclosedAttribute(
+        "is Connect.Profile access token available",
+        true,
+      );
```

Comment:
I just think about that now but we could remove this code from here and put it in the `initProfileClient` client which could be change to `(span: Span) => { addressClient:  ConnectProfileClient, profileClient:  ConnectProfileClient }` or smth like that (but maybe not in this PR, wdyt?)

Reply:
You mean, with all cookies related things ? 🤔

Reply:
Yep, it is redundant and we repeat exactly the same code in each Profile related endpoint

Reply:
> but it should definitely be done in another PR if we go for that

I agree 😁